src/app/utils/parseAsFullyQualifiedURI.ts

```diff
@@ -1,6 +1,9 @@
 export default function parseAsFullyQualifiedURI(uri: string) {
     let origin = '';
     try {
+        // Normalize multiple slashes to a single slash, but don't affect the initial "http://" or "https://"
+        uri = uri.replace(/([^:])\/{2,}/g, '$1/');
```

Comment:
to not to ignore http prefix you could work with `(new URL('http://google.com////adffd////sadf////////')).pathname`

Reply:
The end result is kinda harder to read with this approach, and the regex is going nowhere